### ISSUE TITLE
support load properties file in unnamed module.

### ISSUE THREAD

**Issue by arjenzhou** [brettwooldridge/HikariCP#1826]
since Java 9 introduces modular system in order to restrict accessing among modules, we cannot read unnamed `resources` folder properties files from named module class.

for example, we cannot read `hikari.properties` from `some.java` using `Class.getResourceAsStream` or `Module.getResourceAsStream`.

> |- java
   |- |- foo.bar
   |- |- |- some.java
   |- |- module-info.java 
   |- resources
   |- |- hikari.properties

we can create same structure for `resources` with `java` and `opens` package `foor.bar` as an alternative, or using `ClassLoader.getResourcreAsStream` for convenience.

**Comment by sivaindev** [comment]
Thank you for the fix. Adding the `ClassLoader.getResourceAsStream(...)` fallback appears to be a practical improvement for the common case where users place `hikari.properties` at the resource root (for example `src/main/resources/hikari.properties`).

While reviewing the change and reproducing the behavior in a small JPMS sample project, I wanted to share one technical nuance that may be helpful for future readers.

The previous implementation relied on:

```java
this.getClass().getResourceAsStream(propertyFileName)
```

For a plain resource name such as:

```text
hikari.properties
```

this performs package-relative lookup. Since `HikariConfig` resides in `com/zaxxer/hikari`, the effective lookup would typically be:

```text
com/zaxxer/hikari/hikari.properties
```

As a result, a root-level resource may not be found unless an absolute path is used. In that regard, the newly added `ClassLoader` fallback is beneficial because it also supports root-level resource resolution.

One additional observation from testing is that `ClassLoader.getResourceAsStream(...)` does not appear to be a universal solution in all JPMS scenarios. In a named module, when a resource is packaged inside a non-open package path, `ClassLoader` lookups for that package-scoped resource may still return `null` until the package is opened in `module-info.java`.

For example:

```java
Some.class.getClassLoader()
    .getResourceAsStream("foo/bar/local.properties")
```

may fail when `foo.bar` is encapsulated, but succeed once:

```java
opens foo.bar;
```

is declared.

Because of that, it may be more precise to frame this change as an important compatibility enhancement for root-level resources (including modular applications), rather than as a complete workaround for all Java 9+ module resource-loading restrictions.

In short, the patch is valuable and addresses a real use case; the only suggestion would be to distinguish between:

1. `Class.getResourceAsStream(...)` path semantics (relative vs absolute lookup), and
2. JPMS package encapsulation effects on package-scoped resources.

Appreciate the improvement and the continued maintenance of the project.
